ladybug/windrose.py

```diff
@@ -0,0 +1,761 @@
+        self._legend_parameters = legend_parameters
+
+    @property
+    def legend(self):
+        """Get the Legend object for this plot"""
+        return self._container.legend
+
+    @property
+    def direction_data_collection(self):
+        """Get the direction data collection assigned to this windrose plot."""
+        return self._direction_data_collection
+
+    @property
+    def analysis_data_collection(self):
+        """Get the analysis data collection assigned to this windrose plot."""
+        return self._analysis_data_collection
+
+    @property
+    def analysis_values(self):
+        """Get the hourly wind analysis values to bin by direction (i.e wind speed)."""
+        return self._analysis_data_collection.values
+
+    @property
+    def direction_values(self):
+        """Get the direction data values in this windrose plot."""
+        return self._direction_data_collection.values
+
+    @property
+    def histogram_data(self):
+        """Get a histogram of wind analysis values binned by the wind directions."""
+        return self._histogram_data
+
+    @property
+    def prevailing_direction(self):
+        """Get the predominant direction of the wind values.
+
+        In case of ties, this property will return the first direction value in the
+        ordered direction values.
+        """
+        if self._prevailing_direction is None:
+            dirvals = sorted(self.direction_values)
+            self._prevailing_direction = dirvals[dirvals.index(max(dirvals))]
+        return self._prevailing_direction
+
+    @property
+    def zero_count(self):
+        """Get the number of wind analysis values equal to zero."""
+        return self._zero_count
+
+    @property
+    def analysis_period(self):
+        """Get the AnalysisPeriod assigned to the hourly plot's data collection."""
+        return self._analysis_data_collection.header.analysis_period
+
+    @property
+    def angles(self):
+        """Get a list of orientation angles used to bin analysis values."""
+        return self._angles
+
+    @property
+    def bin_vectors(self):
+        """Get vectors for orientation intervals."""
+        if self._bin_vectors is None:
+            plot_angles = WindRose._compute_plotting_angles(self.angles)
+            self._bin_vectors = WindRose._bin_vectors_radial(plot_angles)
+        return self._bin_vectors
+
+    @property
+    def zeros_per_bin(self):
+        """Get the number of analysis values that are equal to zero, per orientation bin."""
+        if self._zeros_per_bin is None:
+            self._zeros_per_bin = float(self._zero_count) / (len(self.angles) - 1)
+        return self._zeros_per_bin
+
+    @property
+    def real_freq_max(self):
+        """The maximum frequency of the windrose data."""
+        if self._real_freq_max is None:
+            self._real_freq_max = max([len(d) for d in self.histogram_data])
+        return self._real_freq_max
+
+    @property
+    def _container(self):
```

Comment:
Can we make this property public? I don't think we are gaining anything by making it private but there are things that users will gain by being able to access it. For example, they can get the `value_colors` from the container that relate to the `colored_mesh`.